### PR TITLE
Hookify react-countup

### PR DESCRIPTION
**Pull request by johnrom**
This reimplements react-countup as a hook _first_, with a lightweight component wrapper. It cuts the codebase in almost half.

This PR is based off of #514. so that should be merged first. There should be no breaking changes, but I'd still recommend both PRs use a major version bump.

### CODE REVIEW

**Comment by johnrom** [comment]
Updated with changes from https://github.com/glennreyes/react-countup/pull/514

**Comment by mmarkelov** [comment]
@johnrom i'm okay with this PR. Could you please resolve conflicts?

**Comment by johnrom** [comment]
I'll try and take a look next week.

**Comment by johnrom** [comment]
@mmarkelov your changes since I made this PR were pretty much the opposite of mine. however, I think I was able to resolve all of the conflicts successfully.

File: src/useCountUp.ts [hunk 1]
```diff
@@ -1,82 +1,127 @@
...
+  suffix: '',
+  start: 0,
   startOnMount: true,
+  enableReinitialize: true,
 };
 
 const useCountUp = (props: useCountUpProps) => {
-  const parsedProps = { ...defaults, ...props };
-  const { ref } = parsedProps;
+  const config = useMemo(() => ({ ...DEFAULTS, ...props }), [props]);
```
Comment (mmarkelov) [inline src/useCountUp.ts:24]: I think that we can use destructuring here:
```js
const { ref, startOnMount, enableReinitialize, delay, onStart, onEnd, onPauseResume, onReset, onUpdate, ...countUpProps } = useMemo(() => ({ ...DEFAULTS, ...props }), [props]);

...
  const createInstance = useEventCallback(() => {
    return createCountUpInstance(
      typeof ref === 'string' ? ref : ref.current,
      countUpProps
    )
  });
```

**Comment by mmarkelov** [comment]
@johnrom, thanks a lot for your work! i just left one comment to fix and we will ready to merge it.

**Comment by johnrom** [comment]
I have made the requested change. I also noticed some rules of hooks errors and opened a new PR initializing eslint, and fixing all eslint issues using standard eslint + prettier practices.

**Pull request merged**